include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,405 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides seqan3::view::translate_join.
+ */
+
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/alphabet/aminoacid/translation.hpp>
+#include <seqan3/core/add_enum_bitwise_operators.hpp>
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/container/small_string.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/range/view/deep.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/range/view/translate.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+#include <seqan3/range/container/concept.hpp>
```

Comment:
Which of these headers do you really need? I think container, dna5, aa27 are all not needed...

It seems like the following would do:
```cpp
include <seqan3/core/type_traits/range.hpp>
#include <seqan3/range/container/small_string.hpp>
#include <seqan3/range/detail/random_access_iterator.hpp>
#include <seqan3/range/view/translate.hpp>
#include <seqan3/std/concepts>
#include <seqan3/std/ranges>
```
(just guessing)

Reply:
For what would you need ```<seqan3/core/type_traits/range.hpp>```?
But you are right few were not needed. However, you still need ```view_complement``` and ```view_char_to``` as they are used in the test.

Reply:
> For what would you need <seqan3/core/type_traits/range.hpp>?

`reference_t`

> However, you still need view_complement and view_char_to as they are used in the test.

If you just need them in the test, just include them in the test!

Reply:
Putting the include into the test case does not work .. Is that possible for typed tests? Also the includes of ```view_char_to``` for example are always on top also for other view tests. Is that something that is not desired?

Reply:
> Putting the include into the test case does not work .. Is that possible for typed tests? 

I don't understand what you mean. What doesn't work and why? In general every header should include what it uses and not more. Headers required by tests should only be included by the tests...

> Also the includes of view_char_to for example are always on top also for other view tests.

On top of what? 🤔 😆

Reply:
Oooops, I did not read properly .. Thought those are the includes from the test and was extremely confused. Thought you meant putting the include into the test case itself :laughing: 
Sorry for that .. :smile:

Reply:
That's also why I asked for what you would need the type traits ......... :see_no_evil: